src/compiler/parser.ts

```diff
@@ -1637,8 +1637,8 @@ namespace ts {
         // with magic property names like '__proto__'. The 'identifiers' object is used to share a single string instance for
         // each identifier in order to reduce memory consumption.
         function createIdentifier(isIdentifier: boolean, diagnosticMessage?: DiagnosticMessage, privateIdentifierDiagnosticMessage?: DiagnosticMessage): Identifier {
-            identifierCount++;
             if (isIdentifier) {
+                identifierCount++;

```

Comment:
We only use identifierCount for diagnostics purposes. Is there a reason to make it conditional?

Reply:
I didn’t think it was super important, but realized it was double-counting invalid identifiers, and it was easy to make it more accurate.